server-spi-private/src/main/java/org/keycloak/userprofile/DefaultUserProfile.java

```diff
@@ -42,14 +43,21 @@
  */
 public final class DefaultUserProfile implements UserProfile {
 
+    protected final UserProfileMetadata metadata;
     private final Function<Attributes, UserModel> userSupplier;
     private final Attributes attributes;
     private final KeycloakSession session;
     private boolean validated;
     private UserModel user;
 
-    public DefaultUserProfile(Attributes attributes, Function<Attributes, UserModel> userCreator, UserModel user,
+    /**
+     * Contexts where change of email address automatically resets {@link UserModel#EMAIL_VERIFIED} to <code>false</code> 
+     */
+    protected static final Set<UserProfileContext> RESET_EMAIL_VERIFIED_CONTEXTS = new HashSet<>(Arrays.asList(UserProfileContext.ACCOUNT, UserProfileContext.ACCOUNT_OLD, UserProfileContext.UPDATE_PROFILE));
```

Comment:
I think we finally found a situation where the context should be configured with its own metadata to change specific behavior.

I would change the context enum to provide a method `isResetEmailVerified` instead of using this construct. We might need in the future to have more metadata associated with contexts.

Reply:
makes sense, PR updated